User_Story#1/SortingApp/BusinessLayer/Utilities/Parser/ArrayParser.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using BusinessLayer.Extensions;
+using BusinessLayer.Utilities.Parser.Interfaces;
+using System.Collections.Generic;
+
+namespace BusinessLayer.DataParser
+{
+    /// <summary>
+    /// Provides functionality that parses string into array of srecific type.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    /// <typeparam name="T">Certain input type.</typeparam>
+    /// <seealso cref="BusinessLayer.DataParser.Interfaces.IDataParser" />
+    public class ArrayParser<T> : IDataParser<T> where T : IComparable
```

Comment:
No need to specify the constraints here as we already specified them in the interface.